analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -30,6 +32,32 @@ public class NullPointerDereference : NullPointerDereferenceBase
 
         protected override DiagnosticDescriptor Rule => S2259;
 
-        public override bool ShouldExecute() => true;    // ToDo: Implement rule-specific logic
+        public override bool ShouldExecute()
+        {
+            var walker = new SyntaxKindWalker();
+            walker.SafeVisit(Node);
+            return walker.Result;
+        }
+
+        private sealed class SyntaxKindWalker : SafeCSharpSyntaxWalker
+        {
+            public bool Result { get; private set; }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (node.IsAnyKind(
+                    SyntaxKind.AwaitExpression,
+                    SyntaxKind.ElementAccessExpression,
+                    SyntaxKind.ForEachStatement,
+                    SyntaxKind.SimpleMemberAccessExpression))
```

Comment:
I would add the throw expression, even though we don't support it yet.

Reply:
Please add InvocationExpression and tests:
```cs
Action a1 = null;
a1();  // Noncompliant

Action a2 = () => { };
(condition ? a1 : a2)();  // Noncompliant
```

Reply:
Throw should be added once we add support for it.

Reply:
Invocations are not supported, I've added UTs to document FNs

Reply:
Do we also have a ticket? If not, create one.